app/interactions/draw.tsx

```diff
@@ -13,25 +13,30 @@ var vectorSource = new ol.source.Vector({wrapX: false});
 
 export class Draw extends React.Component<any, any> {
   constructor(props) {
-    super(props)
+    super(props);
+    this.state = {
+      zoom: 4,
+      center: [-11000000, 4600000],
+    };
+    this.drawend = this.drawend.bind(this);
   }
 
   drawend(e) {
+    this.setState({ center: [0,0] });
     console.log('xxxxxxxxxxxxx, draw end', e);
   }
-  
+
   render() {
     var typeSelect = document.getElementById('type');
-
     return (
       <div>
-        <Map view={{center: [-11000000, 4600000], zoom: 4}}>
+        <Map view={{center: [-11000000, 4600000], zoom: 4}} setCenter={ this.state.center }>
```

Comment:
Am I the only one who sees how redundant this is? What keeps you from watching the view prop itself? And if you'll insist, why not remove the view prop, replace it with center and zoom, and update accordingly inside `componentWillReceiveProps`?